### PR TITLE
update photon k-factor and lheVProcessor

### PR DESCRIPTION
**Pull request by siqiyyyy**

### CODE REVIEW

File: bucoffea/helpers/gen.py [hunk 1]
```diff
@@ -242,8 +242,9 @@ def setup_gen_jets(df):
         df['nGenJet'],
         pt=df['GenJet_pt'],
         eta=df['GenJet_eta'],
+        abseta=np.abs(df['GenJet_eta']),
         phi=df['GenJet_phi'],
-        mass=0*df['GenJet_pt']
```
Comment (AndreasAlbert) [inline bucoffea/helpers/gen.py]: is this change on purpose? What's it for?

File: bucoffea/gen/lheVProcessor.py [hunk 2]
```diff
@@ -149,6 +158,9 @@ def process(self, df):
                                     )
 
             mask_monojet = monojet_sel.all(*monojet_sel.names)
+            # Apply manual DR>0.4 Cut on GJets sample
+            if is_lo_g(dataset) or is_nlo_g(dataset) or is_lo_g_ewk(dataset) or is_nlo_g_ewk(dataset):
+                mask_monojet = mask_monojet * (df[f'minDR_lhe_parton_photon']>0.4)
```
Comment (AndreasAlbert) [inline bucoffea/gen/lheVProcessor.py]: Instead of manually changing the mask, can't we rather do something like:
```python
if photon:
	monojet_sel.add("lhedr", DR > 0.4)
```
inside of the `monojet_selection` function? Also, same for VBF

**Review by AndreasAlbert — CHANGES_REQUESTED** [review]

File: bucoffea/helpers/gen.py (hunk 1 above)
Comment (siqiyyyy) [reply bucoffea/helpers/gen.py]: No, I don't think the jet mass is used anywhere. I can remove this if you think 0 is better

File: bucoffea/gen/lheVProcessor.py (hunk 2 above)
Comment (siqiyyyy) [reply bucoffea/gen/lheVProcessor.py]: I just thought this is less lines of code. Otherwise we need to add it to both monojet and VBF selections and add DR as an input to the selector

**Comment by siqiyyyy** [comment]
I found this is quite out-dated, should we close this PR?
